code/modules/mob/living/simple_animal/hostile/mining_mobs.dm

```diff
@@ -640,4 +640,134 @@ obj/item/asteroid/basilisk_hide/New()
 		else
 			qdel(A)
 		return
-	return ..()
\ No newline at end of file
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/rockernaut
+	name = "rockernaut"
+	desc = "While a rolling stone may gather no moss, a spinning asteroid may gather semi-sentient moss in the form of a Rockernaut infestation."
+	icon_state = "rocknormal"
+	icon_living = "rocknormal"
+	icon_aggro = "rockcrikey"
+	icon_attack = "rocknormal_to_crikey"
+	icon_attack_time = 5
+	environment_smash_flags = SMASH_LIGHT_STRUCTURES | SMASH_CONTAINERS | SMASH_WALLS
+	attack_sound = 'sound/weapons/heavysmash.ogg'
+	move_to_delay = 20
+	melee_damage_lower = 30
+	melee_damage_upper = 30
+	maxHealth = 350
+	health = 350
+	vision_range = 7
+	speed = 4
+	var/possessed_ore
+
+/mob/living/simple_animal/hostile/asteroid/rockernaut/Life()
+	.=..()
+	if(!.)
+		return 0
+
+	if(stance == HOSTILE_STANCE_IDLE && !client)
+		var/list/can_see = view(get_turf(src), vision_range/2)
+
+		for(var/turf/unsimulated/mineral/M in can_see)
+			if(!M.mineral)
+				continue
+			if(M.rockernaut)
+				continue
+			if(Adjacent(M))
+				//Climb in
+				visible_message("<span class = 'warning'>\The [src] burrows itself into \the [M]!</span>")
+				M.rockernaut = istype(src, /mob/living/simple_animal/hostile/asteroid/rockernaut/boss)?2:TRUE
+				qdel(src)
+				return
+			else
+				if(prob(30))
+					step_towards(src, M)//Step towards it
+					if(environment_smash_flags & SMASH_LIGHT_STRUCTURES)
+						EscapeConfinement()
+				break
+
+
+/mob/living/simple_animal/hostile/asteroid/rockernaut/death()
+	..()
+	visible_message("<span class = 'warning'>\The [src] collapses into a mound of loose rock[possessed_ore?", revealing glittering ore within!":"."]</span>")
+	drop_loot()
+	qdel(src)
+
+/mob/living/simple_animal/hostile/asteroid/rockernaut/proc/drop_loot()
+	if(possessed_ore)
+		for(var/i = 0 to rand(3,9))
+			new possessed_ore(src.loc)
+
+	if(prob(1))
+		new /obj/item/weapon/vinyl/rock(src.loc) //It is a rock monster after all
+
+	for(var/i = 0 to rand(0,3))
+		new /obj/item/weapon/strangerock(src.loc, new /datum/find(get_random_digsite_type(), 0))
+	new /obj/structure/boulder(src.loc)
+
+/mob/living/simple_animal/hostile/asteroid/rockernaut/boss
+	name = "Angie"
+	size = SIZE_HUGE
+	maxHealth = 900
+	move_to_delay = 60
+	health = 900
+	pixel_y = 16 * PIXEL_MULTIPLIER
+	melee_damage_lower = 35
+	melee_damage_upper = 50
+	ranged = 1
+	var/charging = 0
+
+/mob/living/simple_animal/hostile/asteroid/rockernaut/boss/New()
+	..()
+	appearance_flags |= PIXEL_SCALE
+	var/matrix/M = matrix()
+	M.Scale(2,2)
+	transform = M
+
+/mob/living/simple_animal/hostile/asteroid/rockernaut/boss/drop_loot()
```

Comment:
So let me get this straight. A boss monster that only spawns once per roid that has 900 health has a possible loot pool of: 24 to 46 ore of it's type (5 to 9 walls worth of ore good if diamond but meh if anything else), a rock vinyl (very unique loot but not very useful to a miner/xenoarch), and 5 to 13 RANDOM xenoarch finds or if you're unlucky strange rocks that have an even lower chance of being good.
Only the first thing really interests miners and a xenoarch might like the first and last items but the latter will likely end up as junk since it still relies on the RNG of xenoarch finds.

Reply:
A piece of loot that would be easy to code and might be interesting is a hive hand type thing that you can wear that launches boulders in place of the stingers that deal brute. It could be like the borer hammer arm for melee. (slower but hits like a truck)

Reply:
Added some unique loot that would interest miners: Fists of the rockernaut.

Put them on as gloves, and punch through the asteroid!